dashboard/src/store/actions.ts

```diff
@@ -1,15 +1,19 @@
-import Store from "vuex";
 import { System } from "../types/System";
 
 export const actions = {
   async fetchSystems(context: any) {
+    context.commit("setLoading");
     // @ts-expect-error
     const token = await this._vm.$auth.getTokenSilently();
     const systemlist = await fetch("/api/systems/", {
       headers: new Headers({
         Authorization: `Bearer ${token}`
       })
     }).then(response => response.json());
-    context.commit("updateSystemsList", systemlist as Array<System>);
+    const systemMapping: Record<string, System> = {};
+    for (const system of systemlist) {
+      systemMapping[system.system_id] = system as System;
```

Comment:
api has changed so this should be 
```suggestion
      systemMapping[system.object_id] = system.definition as System;
```
or whatever for typescript etc